tests/components/shelly/test_binary_sensor.py

```diff
@@ -39,15 +39,16 @@ async def test_block_binary_sensor(
     entity_id = f"{BINARY_SENSOR_DOMAIN}.test_name_channel_1_overpowering"
     await init_integration(hass, 1)
 
-    assert hass.states.get(entity_id).state == STATE_OFF
+    assert (state := hass.states.get(entity_id))
```

Comment:
Assignments in asserts are generally discouraged

https://docs.astral.sh/ruff/rules/assignment-in-assert/

Reply:
I am a little bit mixed about what is right here. Ruff explanation makes sense for production code since it may cause the assert to skip a required check (" Python interpreter is run under the -O flag, assert statements are not executed."), however it has no effect on tests, since they are based on asserts anyhow and will not work with the `-O`.

They also mention that as long as the variable is used outside the `assert` it will not trigger the rule:
"The rule avoids flagging named expressions that define variables which are only referenced from inside assert statements",
We have no cases here that the extracted variable is used directly inside the assert, but we always use it on another assert on the next statements.